packages/cli/commands/project/watch.js

```diff
@@ -0,0 +1,268 @@
+const chokidar = require('chokidar');
+const path = require('path');
+const { default: PQueue } = require('p-queue');
+const {
+  addAccountOptions,
+  addConfigOptions,
+  setLogLevel,
+  getAccountId,
+  addUseEnvironmentOptions,
+} = require('../../lib/commonOpts');
+const { trackCommandUsage } = require('../../lib/usageTracking');
+const { logDebugInfo } = require('../../lib/debugInfo');
+const {
+  loadConfig,
+  validateConfig,
+  checkAndWarnGitInclusion,
+} = require('@hubspot/cli-lib');
+const {
+  logApiErrorInstance,
+  ApiErrorContext,
+} = require('@hubspot/cli-lib/errorHandlers');
+const { logger } = require('@hubspot/cli-lib/logger');
+const { isAllowedExtension } = require('@hubspot/cli-lib/path');
+const { shouldIgnoreFile } = require('@hubspot/cli-lib/ignoreRules');
+const {
+  provisionBuild,
+  uploadFileToBuild,
+  queueBuild,
+} = require('@hubspot/cli-lib/api/dfs');
+const { validateAccount } = require('../../lib/validation');
+const {
+  getProjectConfig,
+  validateProjectConfig,
+  pollBuildStatus,
+} = require('../../lib/projects');
+const { i18n } = require('@hubspot/cli-lib/lib/lang');
+
+const i18nKey = 'cli.commands.project.subcommands.watch';
+
+const loadAndValidateOptions = async options => {
+  setLogLevel(options);
+  logDebugInfo(options);
+  const { config: configPath } = options;
+  loadConfig(configPath, options);
+  checkAndWarnGitInclusion();
+
+  if (!(validateConfig() && (await validateAccount(options)))) {
+    process.exit(1);
+  }
+};
+
+const queue = new PQueue({
+  concurrency: 10,
+});
+const standbyeQueue = [];
+let timer;
+
+const processStandByQueue = async (accountId, projectName) => {
+  await currentBuild.update(accountId, projectName);
+  queue.addAll(
+    standbyeQueue.map(({ filePath, remotePath }) => {
+      return async () => {
```

Comment:
You might be able to DRY things up a little between this chunk of code and [this one](https://github.com/HubSpot/hubspot-cli/pull/621/files#diff-058f6fead69eab728e1c40ba17719a33d3cac89af8ef94de43e6cf127e90824fR164).